src/Microsoft.ML.DataView/Contracts.cs

```diff
@@ -0,0 +1,50 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// We want the conditional code in this file to always be available to
+// client assemblies that might be DEBUG versions. That is, if someone uses
+// the release build of this assembly to build a DEBUG version of their code,
+// we want Contracts.Assert to be fully functional for that client.
+
+using System;
+
+namespace Microsoft.ML.Internal.DataView
```

Comment:
>DataView [](start = 32, length = 8)

Ah interesting. We can do this later, but if we're doing another "local contracts" we probably ought to add it to the analysis rules here.

[https://github.com/dotnet/machinelearning/blob/db4ecc0135baffc8930201a85fb3e9a101cdfa46/tools-local/Microsoft.ML.InternalCodeAnalyzer/ContractsCheckAnalyzer.cs#L201-L202](https://github.com/dotnet/machinelearning/blob/db4ecc0135baffc8930201a85fb3e9a101cdfa46/tools-local/Microsoft.ML.InternalCodeAnalyzer/ContractsCheckAnalyzer.cs#L201-L202)

But this can be done later... it is more important that we get the shape of the API right at this point.

Reply:
Let me see if I can add it quick.